exotica_core/src/kinematic_tree.cpp

```diff
@@ -957,15 +953,15 @@ void KinematicTree::ResetJointLimits()
 void KinematicTree::UpdateJointLimits()
 {
     joint_limits_.setZero();
-    for (int i = 0; i < num_controlled_joints_; ++i)
+    for (size_t i = 0; i < num_controlled_joints_; ++i)
     {
         joint_limits_(i, 0) = controlled_joints_[i].lock()->joint_limits[0];
         joint_limits_(i, 1) = controlled_joints_[i].lock()->joint_limits[1];
     }
 
     // Update random state distributions for generating random controlled states
     random_state_distributions_.clear();
-    for (unsigned int i = 0; i < num_controlled_joints_; ++i)
+    for (size_t i = 0; i < num_controlled_joints_; ++i)
```

Comment:
This should be `int` to match the data type of `num_controlled_joints_`. `unsigned int` was a mistake.
Happy to merge after that.

Reply:
Why is `num_controlled_joints_` signed anyway? I've not seen a condition where it will have a negative value.

Reply:
According to Stroustrup `unsigned` should only be used if the overflow handling is required or if the API requires it. Various libraries use `unisgned` incorrectly for legacy reasons, including `std`.

Reply:
Well, that's interesting :-) I always use unsigned variables for indexing since I do not see the semantic meaning of negative sizes.
Anyway, I changed the loop variables back to `int`.